modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java

```diff
@@ -1703,6 +1704,68 @@ void exceptionIsThrownIfCommandIsUnknown() {
                 IllegalArgumentException.class,
                 () -> manager.execute(command)
         );
+
+        assertThrows(
+                IllegalArgumentException.class,
+                () -> manager.execute(List.of(
+                        simpleTable("T1"),
+                        simpleTable("T2"),
+                        simpleTable("T3"),
+                        command,
+                        simpleTable("T4")
+                ))
+        );
+    }
+
+    @Test
+    void bulkCommandEitherAppliedAtomicallyOrDoesntAppliedAtAll() {
+        String tableName1 = "TEST1";
+        String tableName2 = "TEST2";
+        String tableName3 = "TEST1"; // intentional name conflict with table1
+
+        List<CatalogCommand> bulkUpdate = List.of(
+                simpleTable(tableName1),
+                simpleTable(tableName2),
+                simpleTable(tableName3)
+        );
+
+        assertThat(manager.table(tableName1, Long.MAX_VALUE), nullValue());
+        assertThat(manager.table(tableName2, Long.MAX_VALUE), nullValue());
+        assertThat(manager.table(tableName3, Long.MAX_VALUE), nullValue());
+
+        assertThat(manager.execute(bulkUpdate), willThrowFast(TableExistsValidationException.class));
+
+        // now let's truncate problematic table and retry
+        assertThat(manager.execute(bulkUpdate.subList(0, bulkUpdate.size() - 1)), willCompleteSuccessfully());
+
+        assertThat(manager.table(tableName1, Long.MAX_VALUE), notNullValue());
+        assertThat(manager.table(tableName2, Long.MAX_VALUE), notNullValue());
+    }
+
+    @Test
+    void bulkUpdateIncrementsVersionByOne() {
```

Comment:
do we need such a test for erroneous operation ? How it was demonstrated in _bulkCommandEitherAppliedAtomicallyOrDoesntAppliedAtAll_ ?

Reply:
did you mean test on bulk operations in case of error does not increment catalog version?

Reply:
yep, i check and seems there is no such a tests

Reply:
test has been added